polyfactory/value_generators/complex_types.py

```diff
@@ -36,26 +36,31 @@ def handle_collection_type(field_meta: FieldMeta, container_type: type, factory:
         return container
 
     if issubclass(container_type, MutableMapping) or is_typeddict(container_type):
-        key = handle_complex_type(field_meta.children[0], factory)
-        value = handle_complex_type(field_meta.children[1], factory)
-        container[key] = value
-
-    elif issubclass(container_type, MutableSequence):
-        container.append(handle_complex_type(field_meta.children[0], factory))
+        key_field_meta: FieldMeta
```

Comment:
this is not the way to type these. use `cast()` on the zip, or cast() the values inside the for loop.